hnn_core/network.py

```diff
@@ -1127,42 +1127,76 @@ def _instantiate_drives(self, tstop, n_trials=1):
-        cell_type : str
-            The cell type whose cells will get the tonic input.
-            Valid inputs are those in `net.cell_types`.
-        amplitude : float
-            The amplitude of the input.
+        cell_types : str | None
+            The name of the cell type to add a tonic input. When supplied,
+            a float value must be provided with the `amplitude` keyword.
+            Valid inputs are those listed in  `net.cell_types`.
+        amplitude: dict | float
+            A dictionary of cell type keys (str) to amplitude values (float).
+            Valid inputs for cell types are those listed in `net.cell_types`.
+            If `cell_types` is not None, `amplitude` should be
+            a float indicating the amplitude of the tonic input
+            for the specified cell type.
         t0 : float
             The start time of tonic input (in ms). Default: 0 (beginning of
-            simulation).
+            simulation). This value will be applied to all the  tonic biases if
+            multiple are specified with the `amplitude` keyword.
         tstop : float
             The end time of tonic input (in ms). Default: end of simulation.
+            This value will be applied to all the  tonic biases if
+            multiple are specified with the `amplitude` keyword.
         """
-        if (cell_type is None or amplitude is None):
-            raise ValueError('cell_type and amplitude must be defined'
-                             f', got {cell_type}, {amplitude}')
-        if 'tonic' not in self.external_biases:
-            self.external_biases['tonic'] = dict()
-        if cell_type in self.external_biases['tonic']:
-            raise ValueError(f'Tonic bias already defined for {cell_type}')
-        if t0 is None:
-            t0 = 0
-        if cell_type not in self.cell_types:
-            raise ValueError(f'cell_type must be one of '
-                             f'{list(self.cell_types.keys())}. '
-                             f'Got {cell_type}')
-
-        self.external_biases['tonic'][cell_type] = {
-            'amplitude': amplitude,
-            't0': t0,
-            'tstop': tstop
-        }
+
+        # old functionality single cell type - amplitude
+        if (cell_type is not None):
+            warnings.warn('cell_type argument will be deprecated and '
+                          'removed in future releases', DeprecationWarning,
+                          stacklevel=1)
+            if (not isinstance(amplitude, float)):
+                raise ValueError('amplitude parameter is not float')
+
+            self._add_cell_type_bias(cell_type=cell_type, amplitude=amplitude,
+                                     t_0=t0, t_stop=tstop)
+        else:
+            if (not isinstance(amplitude, dict)):
+                raise ValueError('amplitude parameter is not a dictionary')
+
+            if (len(amplitude) == 0):
+                warnings.warn('No bias have been defined, no action taken',
```

Comment:
add test for this warning as well

Reply:
it's a bit of a strange warning to raise ... when would `len(amplitude)` have 0 length?

Reply:
Hm. Maybe from a GUI standpoint it could happen if you add a tonic bias "drive" but don't edit any of the amplitude? Though seems like that warning/logic would be better handled in the GUI and not the API.

Reply:
yeah, sounds like something that should be handled in the GUI. Perhaps this speaks to Nick's point of having names for biases as well ... but that may change existing code. I would vote for allowing this "bug" as long as the user can see the list of all added biases in the GUI

Reply:
this wasn't removed? could you address this in the GUI PR?